go/libraries/doltcore/sqle/lookupops/lookupop.go

```diff
@@ -0,0 +1,159 @@
+		return true, nil
+	}
+	// neverContinueRangeCheck will immediately end.
+	neverContinueRangeCheck noms.InRangeCheck = func(tuple types.Tuple) (bool, error) {
+		return false, nil
+	}
+)
+
+// NOTE: The key is a tuple representing a partial or full key. The more values that are in the key, the more specific
+// it is. For example, a full key may have two values, such as [7,8] (ignoring tags). Iterating with the partial key [7]
+// will match all keys starting with [7], including the full key [7,8]. If both operations involving the aforementioned
+// keys are "Equals", then we can say that Equals[7,8] is a subset of Equals[7], since it has less potential matches
+// (1 in this case since the key is full with two values). If we instead have Equals[6] and LessThan[7,8], then
+// Equals[6] is a subset of LessThan[7,8], as all keys starting with the partial key [6] are less than any key matching
+// [7,8]. We cannot, however, say that Equals[7] is a subset of LessThan[7,8], as Equals[7] matches both [7,8] and any
+// key larger, such as [7,9]. They do overlap, from [7,min] to [7,8].
+//
+// Each comparison may have a small comment similar to the above, or an example of a comparison using the standard
+// comparison signs (==, <, >, etc.) suffixed by a tuple.
+
+// NOTE: The function names all contain underscores to make it visually easier to recognize each function considering
+// the length of their names.
+
+// Returns a new LookupOp representing the result set of all possible values.
+func NewLookupOp_All(nbf *types.NomsBinFormat) LookupOp {
+	return &all{
+		nbf: nbf,
+	}
+}
+
+// Returns a new LookupOp representing the empty result set.
+func NewLookupOp_None(nbf *types.NomsBinFormat) LookupOp {
+	return &none{
+		nbf: nbf,
+	}
+}
+
+// Returns a new LookupOp representing the result set (x == key).
+func NewLookupOp_Equals(nbf *types.NomsBinFormat, key types.Tuple) LookupOp {
+	return &equals{
+		nbf: nbf,
+		key: key,
+	}
+}
+
+// Returns a new LookupOp representing the result set (x > key).
+func NewLookupOp_GreaterThan(nbf *types.NomsBinFormat, key types.Tuple) LookupOp {
+	return &greaterThan{
+		nbf: nbf,
+		key: key,
+	}
+}
+
+// Returns a new LookupOp representing the result set (x >= key).
+func NewLookupOp_GreaterThanOrEqual(nbf *types.NomsBinFormat, key types.Tuple) LookupOp {
+	return &greaterThanOrEqual{
+		nbf: nbf,
+		key: key,
+	}
+}
+
+// Returns a new LookupOp representing the result set (x < key).
+func NewLookupOp_LessThan(nbf *types.NomsBinFormat, key types.Tuple) LookupOp {
+	return &lessThan{
+		nbf: nbf,
+		key: key,
+	}
+}
+
+// Returns a new LookupOp representing the result set (x <= key).
+func NewLookupOp_LessThanOrEqual(nbf *types.NomsBinFormat, key types.Tuple) LookupOp {
+	return &lessThanOrEqual{
+		nbf: nbf,
+		key: key,
+	}
+}
+
+// Returns a new LookupOp representing the result set (lt > x > gt).
+func NewLookupOp_GreaterThan_LessThan(nbf *types.NomsBinFormat, gt types.Tuple, lt types.Tuple) LookupOp {
```

Comment:
Would call these Between, and potentially combine a bunch them with the addition of inclusiveStart, inclusiveEnd flags

Reply:
I’ll look into these! And completely forgot about `setalgebra` as well.